### PR TITLE
Duration doc

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @frewsxcv

### CODE REVIEW

File: src/libstd/time/duration.rs [hunk 1]
```diff
@@ -81,26 +109,46 @@ impl Duration {
...
     /// This method does **not** return the length of the duration when
     /// represented by nanoseconds. The returned number always represents a
     /// fractional portion of a second (i.e. it is less than one billion).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::time::Duration;
+    ///
+    /// let five_seconds = Duration::from_millis(5010);
```
Comment (frewsxcv) [inline src/libstd/time/duration.rs]: This is not *exactly* five seconds so the variable binding name is *slightly* misleading. Could also just call it something generic like `duration`. Not a super big deal.

File: src/libstd/time/duration.rs [hunk 2]
```diff
@@ -136,9 +184,11 @@ impl Duration {
         }
     }
 
-    /// Checked duration subtraction. Computes `self + other`, returning `None`
+    /// Checked `Duration` subtraction. Computes `self - other`, returning [`None`]
```
Comment (frewsxcv) [inline src/libstd/time/duration.rs]: Glad you fixed the `+` → `-` error here :)

**Review by frewsxcv — APPROVED** [review]
r=me unless you want to address my comment

File: src/libstd/time/duration.rs [hunk 3]
```diff
@@ -207,9 +259,11 @@ impl Duration {
         }
     }
 
-    /// Checked duration division. Computes `self / other`, returning `None`
+    /// Checked `Duration` division. Computes `self / other`, returning [`None`]
     /// if `other == 0` or the operation results in underflow or overflow.
```
Comment (nagisa) [inline src/libstd/time/duration.rs]: Drive-by fix: division cannot underflow or overflow in `other != Duraton::from_secs(0)` as having a negative or floaty duration is not possible.

File: src/libstd/time/duration.rs [hunk 4]
```diff
@@ -172,8 +222,10 @@ impl Duration {
         }
     }
 
-    /// Checked duration multiplication. Computes `self * other`, returning
-    /// `None` if underflow or overflow occurred.
+    /// Checked `Duration` multiplication. Computes `self * other`, returning
+    /// [`None`] if underflow or overflow occurred.
```
Comment (nagisa) [inline src/libstd/time/duration.rs]: Drive-by fix: multiplication cannot underflow as negative durations are impossible.

File: src/libstd/time/duration.rs [hunk 5]
```diff
@@ -14,17 +14,21 @@ const NANOS_PER_SEC: u32 = 1_000_000_000;
...
 ///
-/// Each duration is composed of a number of seconds and nanosecond precision.
+/// Each `Duration` is composed of a number of seconds and nanosecond precision.
 /// APIs binding a system timeout will typically round up the nanosecond
 /// precision if the underlying system does not support that level of precision.
 ///
-/// Durations implement many common traits, including `Add`, `Sub`, and other
-/// ops traits. Currently a duration may only be inspected for its number of
+/// `Duration`s implement many common traits, including [`Add`], [`Sub`], and other
+/// [`ops`] traits. Currently a `Duration` may only be inspected for its number of
```
Comment (GuillaumeGomez) [inline src/libstd/time/duration.rs]: I'll remove it then.

**Comment by GuillaumeGomez** [comment]
Updated.
